### ISSUE TITLE
E3SM Provenance: Need to check archival of preview_run.log and GIT_* files

### ISSUE THREAD

**Issue by sarats** [ESMCI/cime#4229]
Related issue: https://github.com/ESMCI/cime/issues/3966

There were previously two attempts at recording this information in provenance. However, these files still don't appear in our performance archive. I can see preview_run.log files in a user's run directory but those didn't get archived.
https://github.com/ESMCI/cime/pull/3998
https://github.com/ESMCI/cime/pull/4055

Something to check, the following code snippet from last commit in above PR (https://github.com/ESMCI/cime/pull/4055/commits/b52ce0a07e34e895cd88a8d2545fb5eca7d6c0e0) is _apparently deleting_ the baseline preview_run.log file before making a copy with lid suffix. It should probably make a copy first and then delete. 

https://github.com/ESMCI/cime/blob/master/CIME/provenance.py#L516:L520
```
   if os.path.exists(base_preview_run):
        os.remove(base_preview_run)

    with open(base_preview_run, "w") as fd:
        case.preview_run(lambda x: fd.write("{}\n".format(x)), None)
```

We also need to check why GIT_* files never show up.
https://github.com/ESMCI/cime/blob/master/CIME/provenance.py#L669:675
`        "GIT_STATUS",        "GIT_DIFF",        "GIT_LOG",        "GIT_REMOTE",        "GIT_CONFIG",        "GIT_SUBMODULE_STATUS",
`

**Comment by sarats** [comment]
FYI @rljacob

**Comment by jasonb5** [comment]
I've confirmed this is an issue with @sarats , these missing files are recorded in the build provenance but are not carried over to the performance archive. There is a fix in progress.

**Comment by sarats** [comment]
Please also check if the run_e3sm.sh* script is getting properly saved.

**Comment by jasonb5** [comment]
@sarats Where is `run_e3sm.sh` stored? From the template script it looks like either `${CASE_ROOT}/case_scripts` for production or `${CASE_ROOT}/tests/${run}/case_scripts` for non production, are these paths static? Do we want to archive both cases or only production?

**Comment by sarats** [comment]
@golaz should confirm what he prefers. If I recall correctly, the run script used for a particular job is usually stored in the job's run directory.

**Comment by sarats** [comment]
When I tried the run_e3sm.template.sh, it put a copy of the file in ${CASEROOT}/run_script_provenance/.

**Comment by sarats** [comment]
FYI, I think the CASE_ROOT variable in the script might be different from CIME CASEROOT. 


```
[ac.sarat@chrlogin1 case_scripts]$ pwd
/home/ac.sarat/E3SMv2/sarat_rune3sm-grp/tests/XS_2x5_ndays/case_scripts
[ac.sarat@chrlogin1 case_scripts]$ ./xmlquery CASEROOT
        CASEROOT: /home/ac.sarat/E3SMv2/sarat_rune3sm-grp/tests/XS_2x5_ndays/case_scripts
[ac.sarat@chrlogin1 case_scripts]$ ls run_script_provenance/
run_e3sm.template.sh.20220429-124558
```

**Comment by golaz** [comment]
@sarats : archiving script for production runs should be enough.

**Comment by jasonb5** [comment]
@sarats I've confirmed that the run script is copied into `run_script_provenance` which is created inside the `CIME` case root.

**Comment by rljacob** [comment]
In the run script, CASE_ROOT is the directory where everything goes:  the CIME CASEROOT, the RUNDIR,  EXEROOT, etc.   In normal use, the CIME CASEROOT is named the same as CASENAME but run_e3sm changes that so that its CASE_ROOT is named after the CASENAME and CIME CASEROOT is a generic "case_scripts".

**Comment by jasonb5** [comment]
@rljacob @sarats Looks like we are copying the `run_e3sm.sh*` script, it's under the `CaseDocs` directory in the performance archive.

To find the script we're looking in the  CIME CASEROOT with the glob `run_script_provenance/*`. Is it worth while to update this glob to something more generic like `**/run_e3sm*.sh` in case we ever change where this script gets saved or is the `run_script_provenance/` static directory sufficient?

**Comment by sarats** [comment]
@jasonb5 I presume you did you do a test run. Let me know the machine and I can see if everything is in place within the performance archive.

**Comment by sarats** [comment]
Just a reminder: we need to apply this fix on maint-2.0 as well once it is complete.

**Comment by jasonb5** [comment]
@sarats I did do a test run and all the GIT_* files, preview_run.log and run_e3sm.sh are all present. I sent you a slack message with the location of the outputs to check. Thanks!

**Comment by sarats** [comment]
@jasonb5 Your experiment got uploaded to PACE as well at https://pace.ornl.gov/exp-details/106118 

I checked and it has the GIT_*, preview_run.log and the run_e3sm.sh files. The fix looks good.

**Comment by sarats** [comment]
Please apply this fix to maint-2.0 as well.

**Comment by sarats** [comment]
@jasonb5 Could you confirm that in the case of multiple jobs sharing the same case-directory, we would only copy this particular job id's files, right?
